airbyte-server/src/main/java/io/airbyte/server/handlers/DestinationDefinitionsHandler.java

```diff
@@ -137,15 +158,25 @@ public DestinationDefinitionRead createDestinationDefinition(DestinationDefiniti
 
     configRepository.writeStandardDestinationDefinition(destinationDefinition);
 
-    return buildDestinationDefinitionRead(destinationDefinition);
+    return buildDestinationDefinitionReadWithJobInfo(destinationDefinition, JobConverter.getSynchronousJobRead(response));
   }
 
-  public DestinationDefinitionRead updateDestinationDefinition(DestinationDefinitionUpdate destinationDefinitionUpdate)
+  public DestinationDefinitionReadWithJobInfo updateDestinationDefinition(DestinationDefinitionUpdate destinationDefinitionUpdate)
       throws ConfigNotFoundException, IOException, JsonValidationException {
     final StandardDestinationDefinition currentDestination = configRepository
         .getStandardDestinationDefinition(destinationDefinitionUpdate.getDestinationDefinitionId());
-    imageValidator.assertValidIntegrationImage(currentDestination.getDockerRepository(),
-        destinationDefinitionUpdate.getDockerImageTag());
+
+    // Validates that the docker image exists and can generate a compatible spec by running a getSpec
```

Comment:
Should be in a function so these validations aren't repeated.

Reply:
This section got significantly rewritten.  There's still parallel code between source/dest create/update, but they're doing things slightly differently as well.  I don't see much value in further attempts to refactor.